lib/rmg/TreasurePlacer.cpp

```diff
@@ -527,8 +527,8 @@ size_t TreasurePlacer::getPossibleObjectsSize() const
 }
 
 bool TreasurePlacer::isGuardNeededForTreasure(int value)
-{
-	return zone.getType() != ETemplateZoneType::WATER && value > minGuardedValue;
+{// no guard in a zone with "monsters: none", in a water zone and for small treasures
+	return zone.monsterStrength != EMonsterStrength::ZONE_NONE && zone.getType() != ETemplateZoneType::WATER && value > minGuardedValue;
```

Comment:
since that, can we assign monsterStrength = ZONE_NONE for water zone when it's generated?
It would help to get rid of this hardcoded guard check

Reply:
Good point. I'll check the code that generates the extra water zones not explicitly specified in the template json and set monster strength to zone_none there instead of this hardcoded check.
While at it, I'll also try to figure out why the generator crashed when I specified a template zone's type as "water" in the template json. I did that to the central zone on JC and removed the towns from there and it crashed on develop. I mean, there are no actual templates that use zone type: water, but both on develop and on this branch the generator doesn't complain about a wrong value.

Reply:
Its enought that RMG is trying to spawn any objects which don't have a water template. In particular Mines, but also about anything without HoTA.

Reply:
Did as you suggested, Nordsoft, and also updated the branch with recent changes from develop.